cfg/libcerror.cfg

```diff
@@ -0,0 +1,128 @@
+<?xml version="1.0"?>
+<!-- Official repository: https://github.com/libyal/libcerror -->
+<!-- Typically included by: #include <libcerror.h> -->
+<def format="2">
+  <!-- int libcerror_error_initialize( libcerror_error_t **error,
+                                       int error_domain,
+                                       int error_code ); -->
+  <function name="libcerror_error_initialize">
+    <noreturn>false</noreturn>
+    <returnValue type="int"/>
```

Comment:
Would you like to add the specification “`<use-retval/>`” for this function description?

Reply:
Not as long as we want to avoid false positive as good as possible.
One does not have to check the return value. It is also possible to check if `*error` is not NULL if it has been initialized to NULL before to know that it was not possible to allocate the memory for example.

Reply:
>One does not have to check the return value.

I disagree to such a view because the mentioned functions can indicate by their return values if a call failed.

Reply:
Feel free to create a PR to change it. I am not sure if it gets accepted. Personally i do not have problems to specify it with `<use-retval/>`, but others have to agree with it also.

Reply:
I am against it because this suggestion comes from @elfring. if @elfring tried to follow the cppcheck philosophy I would listen more but right now I will not listen to any advice.

Reply:
>I am against it because this suggestion comes from @elfring.

* I find such a reason questionable.
  Specific change proposals (also from me) were accepted and I hope that related software improvements can happen again.
* I am curious on how the presented communication difficulties can eventually be resolved in more constructives ways.

Reply:
> I find such a reason questionable.

no. you continue making suggestions against the cppcheck philosophy.. so I now consider all these suggestions invalid.

Reply:
>so I now consider all these suggestions invalid.

Do you care for memory allocation failures also with your software development approaches?

Reply:
How do you think about to add the XML element “memory” for this programming interface description?